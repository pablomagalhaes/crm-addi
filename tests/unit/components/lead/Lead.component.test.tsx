import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Lead from '@/components/lead/Lead.component';

jest.mock('@/io/redux/lead/useLead', () =>
  jest.fn(() => ({
    isLoading: false,
    isRefused: false,
    wasFeched: false,
    validateLead: jest.fn(),
  }))
);

const lead = {
  id: '1',
  firstName: 'Ashlyn',
  lastName: 'Reeve',
  nationalIdNumber: '70143939017',
  birthdate: '1990-07-03',
  email: 'ashlynreeve@gmail.com',

  existsInRegisty: null,
  matchWithRegisty: null,
  hasJudicialRecord: null,
  score: 0,
};

describe('<Lead />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<Lead lead={lead} />);
    expect(wrapper).toMatchSnapshot();
  });
});
