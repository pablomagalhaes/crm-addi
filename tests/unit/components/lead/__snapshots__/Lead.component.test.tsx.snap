// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lead /> should match snapchat 1`] = `
<tr>
  <td
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    1
  </td>
  <td>
    Ashlyn Reeve
  </td>
  <td>
    70143939017
  </td>
  <td>
    1990-07-03
  </td>
  <td>
    ashlynreeve@gmail.com
  </td>
  <td
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <RegistreyCell
      existsInRegisty={null}
      isLoading={false}
      matchWithRegisty={null}
    />
  </td>
  <td
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <RegistryCell
      hasJudicialRecord={null}
      isLoading={false}
    />
  </td>
  <td
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <ScoreCell
      isLoading={false}
      score={0}
    />
  </td>
  <td>
    <styled.button
      onClick={[Function]}
    >
      Validate
    </styled.button>
  </td>
</tr>
`;
